### PR TITLE
[Doc] Update distributed chapter according to new pipeline

### PR DESCRIPTION
**Pull request by jermainewang**
## Description

* Replace the old distributed preprocess section with the new pipeline
* Improve other parts
* Fix some API link issues.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

### CODE REVIEW

File: docs/source/guide/distributed-partition.rst [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+.. code-block:: none
+
+    <node_type> <weight1> ... <orig_type_node_id>
+
+All fields are separated by whitespace:
+
+* ``<node_type>`` is an integer. For a homogeneous graph, its value is always
+  0. For heterogeneous graphs, its value indicates the type of each node.
+* ``<weight1>``, ``<weight2>``, etc. are integers that indicate the node weights
+  used by ParMETIS to balance graph partitions. If a user does not provide node
```
Comment (classicsong) [inline docs/source/guide/distributed-partition.rst:197]: The weights here is a little bit confusing. It is more like node groups that ParMETIS will try to balance the number of nodes within a group across different partitions.

File: docs/source/guide/distributed-partition.rst [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+  partition (it is important to balance graph partitions in order to achieve
+  good training speed). However, this default strategy may not be sufficient
+  for many use cases.  For example, in a heterogeneous graph, we want to
+  partition the graph so that all partitions have roughly the same number of
+  nodes for each node type. The toy example below shows how we can use node
+  weights to balance the number of nodes of different types.
+* ``<orig_type_node_id>`` is an integer representing the node ID in its own
+  type. In DGL, nodes of each type are assigned with IDs starting from 0. For a
+  homogeneous graph, this field is the same as the node ID.
+* The row ID indicates the *homogeneous* node IDs.  All nodes of the same type
```
Comment (classicsong) [inline docs/source/guide/distributed-partition.rst:208]: There is no row ID now.

File: docs/source/guide/distributed-partition.rst [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+.. code-block:: none
+
+    <node_id> <node_type> <weight1> ... <orig_type_node_id>
+
+* ``<node_id>`` is the *homogeneous* node IDs after ID reassignment.
+* ``<node_type>`` is the node type.
+* ``<weight1>`` is the node weight used by ParMETIS.
+* ``<orig_type_node_id>`` is the original node ID for a specific node type in the input heterogeneous graph.
+* ``<attributes>`` are optional fields that contain any node attributes in the input node file.
```
Comment (classicsong) [inline docs/source/guide/distributed-partition.rst:278]: The <attributes> is not in the code-block.

File: docs/source/guide/distributed-partition.rst [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+    <src_id> <dst_id> <orig_src_id> <orig_dst_id> <orig_type_edge_id> <edge_type>
+
+* ``<src_id>`` is the *homogeneous* ID of the source node after ID reassignment.
+* ``<dst_id>`` is the *homogeneous* ID of the destination node after ID reassignment.
+* ``<orig_src_id>`` is the *homogeneous* ID of the source node in the input graph.
+* ``<orig_dst_id>`` is the *homogeneous* ID of the destination node in the input graph.
+* ``<orig_type_edge_id>`` is the edge ID for the specific edge type in the input graph.
+* ``<edge_type>`` is the edge type.
+* ``<attributes>`` are optional fields that contain any edge attributes in the input edge file.
```
Comment (classicsong) [inline docs/source/guide/distributed-partition.rst:293]: The is not in the code-block.

File: docs/source/guide/distributed-partition.rst [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+
+.. code-block:: none
+
+    <src_id> <dst_id> <orig_src_id> <orig_dst_id> <orig_type_edge_id> <edge_type>
+
+* ``<src_id>`` is the *homogeneous* ID of the source node after ID reassignment.
+* ``<dst_id>`` is the *homogeneous* ID of the destination node after ID reassignment.
+* ``<orig_src_id>`` is the *homogeneous* ID of the source node in the input graph.
+* ``<orig_dst_id>`` is the *homogeneous* ID of the destination node in the input graph.
+* ``<orig_type_edge_id>`` is the edge ID for the specific edge type in the input graph.
```
Comment (classicsong) [inline docs/source/guide/distributed-partition.rst:291]: Is the edge ID here the one of the specific edge type in the input heterogeneous graph?

Comment (classicsong) [reply docs/source/guide/distributed-partition.rst:291]: I.E. per-type edge ID.

File: docs/source/guide/distributed-partition.rst [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+* ``<edge_type>`` is the edge type.
+* ``<attributes>`` are optional fields that contain any edge attributes in the input edge file.
+
+When invoking ``pm_dglpart``, the three input files: ``xxx_nodes.txt``,
+``xxx_edges.txt``, ``xxx_stats.txt`` should be located in the directory where
+``pm_dglpart`` runs. The following command run four ParMETIS processes to
+partition the graph named ``xxx`` into eight partitions (each process handles
+two partitions).
+
```
Comment (classicsong) [inline docs/source/guide/distributed-partition.rst:300]: Shall we give an example of p<part_id>-xxx_nodes.txt, p<part_id>-xxx_edges.txt, p<part_id>-xxx_stats.txt ?

File: docs/source/guide/distributed-partition.rst (hunk 5 above)
Comment (jermainewang) [reply docs/source/guide/distributed-partition.rst:291]: yes. I changed it to type_wise_edge_id

File: docs/source/guide/distributed-partition.rst (hunk 6 above)
Comment (jermainewang) [reply docs/source/guide/distributed-partition.rst:300]: This may be too complicated. I leave it for now.

**Review by classicsong — APPROVED** [review]

**Pull request merged**